contracts/strategy/rysk/RyskStrategy.sol

```diff
@@ -0,0 +1,358 @@
+            return true;
+
+        IRyskLiquidityPool.WithdrawalReceipt
+            memory withdrawalReceipt = _getWithdrawalReceipt();
+
+        return withdrawalReceipt.shares != 0;
+    }
+
+    /// @inheritdoc IStrategy
+    /// @notice this also includes shares from the pending withdrawal and shares yet to be minted for pending deposits
+    function investedAssets()
+        public
+        view
+        virtual
+        override(IStrategy)
+        returns (uint256)
+    {
+        IRyskLiquidityPool.DepositReceipt
+            memory depositReceipt = _getDepositReceipt();
+        IRyskLiquidityPool.WithdrawalReceipt
+            memory withdrawalReceipt = _getWithdrawalReceipt();
+        uint256 currentPricePerShare = ryskLqPool.withdrawalEpochPricePerShare(
+            ryskLqPool.withdrawalEpoch()
+        );
+
+        // shares for pending deposit (not yet minted or not yet updated on the deposit receipt as unredeemed)
+        uint256 amountInPendingDepositShares = _getAmountInSharesForPendingDeposit(
+                depositReceipt.epoch,
+                depositReceipt.amount,
+                currentPricePerShare
+            );
+
+        // shares marked for withdrawal that are now owned by the pool itself
+        uint256 amountInPendingWithdrawalShares = _sharesToUnderlying(
+            withdrawalReceipt.shares,
+            ryskLqPool.withdrawalEpochPricePerShare(withdrawalReceipt.epoch)
+        );
+
+        // unredeemed shares are not counted in the strategy's balance
+        // because they are not yet claimed (they are still owned by the pool)
+        uint256 amountInUnredeemedShares = _sharesToUnderlying(
+            depositReceipt.unredeemedShares,
+            currentPricePerShare
+        );
+
+        // redeemed shares that are owned by the strategy
+        uint256 amountInRedeemedShares = _sharesToUnderlying(
+            _getSharesBalance(),
+            currentPricePerShare
+        );
+
+        return
+            amountInPendingDepositShares +
+            amountInPendingWithdrawalShares +
+            amountInUnredeemedShares +
+            amountInRedeemedShares;
+    }
+
+    /// @inheritdoc IStrategy
+    function invest() external virtual override(IStrategy) onlyManager {
+        uint256 balance = underlying.balanceOf(address(this));
+
+        if (balance == 0) revert StrategyNoUnderlying();
+
+        emit StrategyInvested(balance);
+        ryskLqPool.deposit(balance);
+    }
+
+    /// @inheritdoc IStrategy
+    function withdrawToVault(uint256 _amount)
+        external
+        virtual
+        override(IStrategy)
+        onlyManager
+    {
+        if (_amount == 0) revert StrategyAmountZero();
+
+        uint256 currentWithdrawalEpoch = ryskLqPool.withdrawalEpoch();
+        _checkPendingWithdrawal(currentWithdrawalEpoch);
+
+        uint256 currentPricePerShare = ryskLqPool.withdrawalEpochPricePerShare(
```

Comment:
this isn't really the currentPricePershare right? There's the pricerPerShare for current withdrawal epoch, and there's the pricerPerShare for the current deposit. I'm not sure calling this `current` is correct. Maybe add a `withdrawal` in there?

Reply:
discussed in person